### ISSUE TITLE
Question about select samples for mix

### ISSUE THREAD

**Issue by wangnaihao** [JackYFL/DISC#2]
感谢您杰出的工作，实例动态阈值让我深受启发！同时，我在阅读您的论文和代码时有一些疑惑，在论文中`self.labels = torch.ones(N).long().to(self.device)`只在第一次加载数据时被初始化为数据集的标签，如果不考虑存在over-confidence的样本，那么实例动态阈值来选择样本是仅仅存在于标签正确的样本部分，也就是说clean set 和 hard set其实是来自标签正确的样本集合的，不知道我的理解是否正确，感谢您的回答！祝科研顺利！

**Comment by JackYFL** [comment]
感谢您对我们工作的关注！这里self.labels实际上加载的是数据中的噪声标签，这样做的目的是DISC在挑选的过程中是从给定噪声标签中进行挑选的，所以我们需要提前获取噪声标注。另外，DISC算法中还有一个定义为gt_labels，这个存储的是正确标签，为了验证训练时挑选或校正的准确性（仅在CIFAR数据集中进行加载）。如果您还有什么其他问题，欢迎向我提问。

**Comment by wangnaihao** [comment]
感谢您的回复！我注意到`w_mask = self.w_probs[self.labels>=0, self.labels]>self.w_prev_confidence[self.labels>=0]`这句代码中self.labels实际上是含噪的标签，因此对于没有被污染的、正确的标签会存在prob随着学习递增的现象，但是对于被污染的噪声标签，`self.w_probs[self.labels>=0, self.labels]`是N*C的一个向量，那么在self.labels位置对应的概率通常会小于后一项的预测概率，所以我在上一个问题中提出clean和hard两个集合其实大部分是来自没有被污染的标签，不知道是不是我对代码或者文章理解有误！再次感谢您的回复！

**Comment by JackYFL** [comment]
您的理解是对的。挑选出来的样本大部分都是正确的，这里主要利用到了单标签分类中的神经网络记忆效应的特性，网络前期倾向于先拟合干净标签，因此前期挑选出来的集合大部分都是正确的，而这种正确的挑选使得网络的决策边界与真实边界更相近，使得训练朝着更好的方向发展：即网络对于正确标签的记忆程度越来越深(confidence高于动态阈值)，而对于错误标签的记忆程度逐渐下降(confidence低于动态阈值)。


| |
李一帆
|
|
***@***.***
|
---- Replied Message ----
| From | ***@***.***> |
| Date | 6/19/2023 14:12 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [JackYFL/DISC] Question about select samples for mix (Issue #2) |

感谢您杰出的工作，实例动态阈值让我深受启发！同时，我在阅读您的论文和代码时有一些疑惑，在论文中self.labels = torch.ones(N).long().to(self.device)只在第一次加载数据时被初始化为数据集的标签，如果不考虑存在over-confidence的样本，那么实例动态阈值来选择样本是仅仅存在于标签正确的样本部分，也就是说clean set 和 hard set其实是来自标签正确的样本集合的，不知道我的理解是否正确，感谢您的回答！祝科研顺利！

**Comment by wangnaihao** [comment]
感谢您的回复！再次祝您科研顺利！

**Comment by JackYFL** [comment]
谢谢，也祝您科研顺利~
